server/container/guice/mailet/src/main/java/org/apache/james/utils/GuiceGenericLoader.java

```diff
@@ -19,24 +19,77 @@
 
 package org.apache.james.utils;
 
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.apache.james.filesystem.api.FileSystem;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.fge.lambdas.Throwing;
+import com.github.steveash.guavate.Guavate;
+import com.google.common.collect.ImmutableList;
 import com.google.inject.Injector;
 
 public class GuiceGenericLoader<T> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GuiceGenericLoader.class);
+    public static final String INCLUDED_JARS_FOLDER_NAME = "included-jars/";
 
     private final Injector injector;
     private final String defaultPackageName;
+    private URLClassLoader urlClassLoader;
 
-    public GuiceGenericLoader(Injector injector, String defaultPackageName) {
+    public GuiceGenericLoader(Injector injector, FileSystem fileSystem, String defaultPackageName) {
         this.injector = injector;
         this.defaultPackageName = defaultPackageName;
+
+        this.urlClassLoader = new URLClassLoader(retrieveIncludedUrls(fileSystem), getClass().getClassLoader());
+    }
+
+    private URL[] retrieveIncludedUrls(FileSystem fileSystem) {
+        try {
+            File file = fileSystem.getFile("file://" + INCLUDED_JARS_FOLDER_NAME);
+            ImmutableList<URL> urls = recursiveExpend(file)
+                .collect(Guavate.toImmutableList());
+            return urls.toArray(new URL[urls.size()]);
```

Comment:
you don't need to know the size: use .toArray(size -> new URL[size])
or I prefer: .toArray(URL[]::new)